tests/gfoidl.Base64.Tests/Internal/Base64EncoderTests/GetDecodedLength.cs

```diff
@@ -19,6 +19,26 @@ public void EncodedLength_is_0___0()
         }
         //---------------------------------------------------------------------
         [Test]
+        public void MaxEncodedSpan_1_to_50_given___correct_decoded_len()
+        {
+            var sut = new Base64Encoder();
+
+            Assert.Multiple(() =>
+            {
+                for (int i = 1; i < 50; ++i)
+                {
+                    var data = new byte[i];
+                    string base64WoPaddingString = Convert.ToBase64String(data);
+                    byte[] base64WoPadding = Encoding.ASCII.GetBytes(base64WoPaddingString);
+                    int padding = i % 3 == 0 ? 0 : 3 - i % 3;
+                    int actual = sut.GetMaxDecodedLength(base64WoPadding.Length);
+
+                    Assert.AreEqual(i + padding, actual);
```

Comment:
It's nice to have the exact length, but we test for _max decoded length_, so `Assert.GreaterOrEqual` would be enough.

Same in base64Url.

But you can leave it as it is.